lib/docker.ts

```diff
@@ -89,6 +89,33 @@ class Docker {
     ]);
   }
 
+  public async pull() {
+    return subProcess.execute("docker", ["pull", this.targetImage]);
+  }
+
+  public async isImageLocal(targetImage: string) {
+    try {
+      await this.inspectImage(targetImage);
+      return true;
+    } catch (err) {
+      return false;
+    }
+  }
+
+  public async inspectWithPull(targetImage: string) {
+    const isImageLocal = await this.isImageLocal(targetImage);
```

Comment:
Hmm, also this is calling a function with more a less a single line of code. I think I would write something like:
```
try {
  this.inspectImage(...
} catch() {
  this.pull()
}
return this.inspectImage(...
```
I suggest creating a `isImageLocal` function (with only a single line of actual code) only if there are at least two callers.

Reply:
Wanted to have a clearer error reflected, but I see your point :)